internal/pkg/helm/values_inspector.go

```diff
@@ -91,3 +100,141 @@ func isBooleanYamlValue(input *map[string]interface{}, path *[]string, pathIndex
+		for i := 0; i < items.Len(); i++ {
+			item := items.Index(i)
+			if item.Elem().Kind() == reflect.String {
+				returnMap[item.Interface().(string)] = []*LogHelmReport{}
+			} else if item.Elem().Kind() == reflect.Map {
+				returnMaps := item.Interface().(map[string]interface{})
+				for k := range returnMaps {
+					returnMap[k] = []*LogHelmReport{}
+				}
+			}
+		}
+	}
+	return &returnMap
+}
+
+//PreFixValuesWithItems...For single item list {% for item in hosts %}{{ item.name }}{% endfor %},
+//This function will prefix the list variables with a loop variable, within the body of the for loop.
+//There is two cases , one with field name that needs to be prefixed with loop variables
+// other condition is dot field name is replaced by loop variable
+func PreFixValuesWithItems(sb *strings.Builder, itemField string, pathValues *map[string][]*LogHelmReport) {
+	input := sb.String()
+	lines := strings.Split(string(input), "\n")
+	lineNumbers := []int{}
+	dotSuffixedField := " " + itemField + goTemplateMemberAccessOperator
+	sb.Reset()
+	for i, line := range lines {
+		for vars := range *pathValues {
+			if vars == goTemplateMemberAccessOperator { // if the replacing variable is dot then do this
+				re, err := regexp.Compile(`\s\.\s`)
+				if err == nil {
+					matched := re.MatchString(line)
+					if matched { //ignore line with template and includes
+						if strings.Contains(line, "template") || strings.Contains(line, "include") {
+							continue
+						}
+						lines[i] = re.ReplaceAllString(line, " "+itemField+" ")
+						logrus.Debugf("Replaced item %s  to : %s on line %d", vars, itemField, i)
+						r := new(LogHelmReport)
+						r.Action = vars + "Replaced with " + itemField + "\n"
+						r.NewLine = lines[i]
+						r.OriginalLine = line
+						r.FieldName = itemField
+						report := (*pathValues)[goTemplateMemberAccessOperator]
+						report = append(report, r)
+						(*pathValues)[goTemplateMemberAccessOperator] = report
+					}
+				}
+			} else {
+				re, err := regexp.Compile("\\s." + vars + "\\s")
+				if err == nil {
+					matched := re.MatchString(line)
+					if matched {
+						lines[i] = re.ReplaceAllString(line, dotSuffixedField+vars+" ")
+						logrus.Debugf("Appending item %s key to : %s on line %d", vars, dotSuffixedField+vars, i)
+						r := new(LogHelmReport)
+						r.Action = vars + " Replaced with " + itemField + "\n"
+						r.NewLine = lines[i]
+						r.OriginalLine = line
+						r.FieldName = itemField
+						report := (*pathValues)[goTemplateMemberAccessOperator]
+						report = append(report, r)
+						(*pathValues)[goTemplateMemberAccessOperator] = report
+					}
+				}
+			}
+			lineNumbers = append(lineNumbers, i)
+		}
+	}
+	if len(lineNumbers) > 0 {
+		logrus.Infof("Successfully updated references for range item field: %s on lines %d",
+			itemField, lineNumbers)
+	}
+	output := strings.Join(lines, "\n")
+	sb.WriteString(output)
+
+}
+
+//Remove $ from loop variables and body of the loop
```

Comment:
Should be:
```
// Remove $ from loop variables and body of the loop
```

Reply:
Same here, We need to enable this rule in golint. I feel the fix should be in the linter